lib/model/query/entities.js

```diff
@@ -163,6 +165,16 @@ const processSubmissionEvent = (event) => (container) =>
           errorMessage: err.message,
           problem: (err.isProblem === true) ? err : null }));
 
+const createEntitiesFromPendingSubmissions = (submissionEvents, parentEvent) => (container) =>
+  // run sequentially because we want to isolate transaction for each submission
+  runSequentially(submissionEvents.map(event =>
+  // what should be eventId here? If we pass event.id then we won't be able to retrieve submission InstanceId when submission is deleted
+  // and if we pass submission create/edit eventId then we are telling a lie. should we store both? how it will look with new source table
```

Comment:
Is this comment question answered by passing the parent `dataset.update` event id and storing it in the source details?